packages/mongo/oplog_v2_converter.js

```diff
@@ -97,6 +96,7 @@ function flattenObject(ob) {
 
     if (typeof ob[i] == 'object' && ob[i] !== null) {
       const flatObject = flattenObject(ob[i]);
+      if(Object.keys(flatObject).length === 0) { return ob; }
       for (const x in flatObject) {
         if (!flatObject.hasOwnProperty(x)) continue;
```

Comment:
I'm not sure how important performance is here or how large the objects are so it could be fine as is, but this loop is duplicating the work done by Object.keys. It would be a little more efficient to store the result of Object.keys in a variable, and then loop over the keys.

Reply:
thanks, @zodern , I've included your suggestion in the other PR I made!